App/Sagas/TextileSagas.ts

```diff
@@ -89,24 +89,38 @@ export function * handleProfilePhotoSelected(action: ActionType<typeof UIActions
   yield take(getType(TextileNodeActions.startNodeSuccess))
 
   let defaultThread: TextileTypes.Thread | undefined = yield call(getDefaultThread)
-    if (!defaultThread) {
-      yield put(ThreadsActions.addThreadRequest('default'))
-      const action: ActionType<typeof ThreadsActions.addThreadSuccess> = yield take(getType(ThreadsActions.addThreadSuccess))
-      defaultThread = action.payload.thread
-      yield put(ThreadsActions.refreshThreadsRequest())
-    }
+  if (!defaultThread) {
+    yield put(ThreadsActions.addThreadRequest('default'))
+    const action: ActionType<typeof ThreadsActions.addThreadSuccess> = yield take(getType(ThreadsActions.addThreadSuccess))
+    defaultThread = action.payload.thread
+    yield put(ThreadsActions.refreshThreadsRequest())
+  }
+  yield processAvatarImage(action.payload.uri, defaultThread)
+}
+
+export function * handleProfilePhotoUpdated(action: ActionType<typeof UIActions.updateProfilePicture>) {
+  yield call(NavigationService.navigate, 'TabNavigator')
+
+  let defaultThread: TextileTypes.Thread = yield call(getDefaultThread)
 
-  const photoPath = action.payload.uri.replace('file://', '')
+  yield processAvatarImage(action.payload.uri, defaultThread)
```

Comment:
Ok probably last thing, this should be `yield * AvatarImage...`. I think you can also use `yield call()` to run other generator functions too. Not sure what the diff is.